openvdb_points/tools/PointConversion.h

```diff
@@ -979,6 +1074,151 @@ convertPointDataGridGroup(  Group& group,
     group.finalize();
 }
 
+template<typename PositionWrapper, typename InterrupterT>
+inline float autoVoxelSize(const PositionWrapper& positions,
+                           const size_t pointsPerVoxel,
+                           InterrupterT* interrupter,
+                           Vec3R minBounds,
+                           Vec3R maxBounds,
+                           const math::Mat4d targetTransform,
+                           const Index decimalPlaces)
+{
+    using namespace point_conversion_internal;
+
+    if(pointsPerVoxel == 0) OPENVDB_THROW(ValueError, "Points per voxel cannot be zero.");
+
+    float voxelSize(0.1f);
+
+    const size_t numPoints = positions.size();
+
+    // return the default voxel size if we have zero or only 1 point
+
+    if(numPoints <= 1) return voxelSize;
+
+    // if the provided bounds are invalid, try to recompute
+
+    if(minBounds[0] > maxBounds[0] ||
+       minBounds[1] > maxBounds[2] ||
+       minBounds[2] > maxBounds[2])
+    {
+        tbb::blocked_range<size_t> range(0, numPoints);
+        CalculatePositionBounds<PositionWrapper> calculateBounds(positions);
+        tbb::parallel_reduce(range, calculateBounds);
+        calculateBounds.getBounds(minBounds, maxBounds);
+    }
+
+    // return default size if points all occupy the same space (on-top of each other)
+
+    if(minBounds == maxBounds) return voxelSize;
+
+    // if point lie along a plane in any direction, make sure we construct a valid volume
+
+    for(size_t i = 0; i < 3; ++i) {
+        if(!math::isApproxEqual(minBounds[i], maxBounds[i])) continue;
+        minBounds[i] -= voxelSize / 2.0f;
+        maxBounds[i] += voxelSize / 2.0f;
+    }
+
+    const Vec3d extents(maxBounds - minBounds);
+    double volume = extents[0] * extents[1] * extents[2];
+    assert(!math::isApproxZero(volume));
```

Comment:
It is possible for the size in each axis to just be valid (ie slightly more than 1e-7) resulting in a very tiny resulting volume  (ie 1e-20) which would produce a huge voxel size. Probably worth handling when the volume is also approx zero. (And perhaps unit test points that only just pass this tolerance).